CODE-GUIDELINES.md

```diff
@@ -37,3 +37,26 @@ test('...', () => {
   vi.mocked(window.windowMethod).mockResolvedValue('a string');
 });
 ```
+
+### Do not use `try {} catch` in tests
+
+The following code has a problem, if the code in the try block do not raise an error, we do not check the error message.
+
+```ts
+// bad
+test('expect error to be properly catched and verified', async () => {
+  try {
+    await Promise.reject(new Error('Dummy Error'));
+  } catch(err: unknown) {
+    expect(err.message).toBe('Dummy Error')
+  }
+});
+```
+
+Here is a version where we are sure the block rejected an error.
+
+```ts
+  await expect(async () => {
+    await Promise.reject(new Error('Dummy Error'));
+  }).rejects.toThrowError('Dummy Error');
+```
```

Comment:
I would not use the Promise.reject inside the code directly but saying we're calling a method that is throwing an error

more like https://vitest.dev/api/expect#rejects

async function erroneousMethod(): Promise<void> {
 ...
  throw new Error('Dummy Error');
 }

 try {
    await erroneousMethod();
  } catch(err: unknown) {
    expect(err.message).toBe('Dummy Error')
  }

Reply:
Thanks, applied